### PR TITLE
✨ Improve `kubestellar init` wrt waiting for kcp

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
## Summary
This PR does two things.
(1) Make waiting for a "kcp" process optional.
(2) Add non-optional waiting for the server to be fully ready.

## Related issue(s)

I think this needs to go in before #1077 so that can use this.

### CODE REVIEW

**Comment by MikeSpreitzer** [comment]
The force-push to 78b243b16: adds timing to the wait messages, and changes the checked workspace to `root:compute`.

**Comment by MikeSpreitzer** [comment]
And the force-push to 810670655 is the doc update.

File: scripts/kubestellar [hunk 1]
```diff
@@ -302,6 +305,21 @@ if [ $subcommand == stop ]; then
...
+# Check if kcp is running
+if [ "$local_kcp" == true ] && [ $(process_running kcp) != "running" ]; then
+    echo "kcp is not running - please start it ...."
+    exit 1
+fi
+
+let tries=1
+while ! ( kubectl get workspaces && kubectl ws root:compute ); do
+    if (( ( tries & (tries-1) ) == 0 ))
+    then echo "$0: At $(date), kcp server not running yet, waiting $(( (tries+2)/2 * 15 )) sec..." >& 2
```
Comment (ezrasilvera) [inline scripts/kubestellar:317]: let's also add "Or root:compute is not available"  to the message

**Comment by MikeSpreitzer** [comment]
Recent force-pushes update the waiting message and wait time arithmetic.

File: scripts/kubestellar [hunk 2]
```diff
@@ -302,6 +305,21 @@ if [ $subcommand == stop ]; then
...
 fi
 
+# Check if kcp is running
+if [ "$local_kcp" == true ] && [ $(process_running kcp) != "running" ]; then
+    echo "kcp is not running - please start it ...."
+    exit 1
+fi
+
+let tries=1
+while ! ( kubectl get workspaces && kubectl ws root:compute ); do
```
Comment (ezrasilvera) [inline scripts/kubestellar:315]: Do we have any test to verify that kubectl ws plugin is in the path ?

Comment (MikeSpreitzer) [reply scripts/kubestellar:315]: well, this is testing that.

**Comment by MikeSpreitzer** [comment]
If we want the Helm chart to just launch pods without any synchronization (which I think we do), then each needs to wait for the preconditions that it needs.

I would be willing to entertain a time limit on the waiting for kcp and root:compute.  That would require deciding on the default time limit.

File: scripts/kubestellar (hunk 2 above)
Comment (ezrasilvera) [reply scripts/kubestellar:315]: OK. I meant distinctly. Fine for now, for the future we may want separate tests for better root cause analysis

**Pull request merged**